recipes/inkscape/wrapper.c

```diff
@@ -0,0 +1,262 @@
+/*
+    Wrapper for Windows to avoid using bat files
+
+    To build/rebuild with mingw32, do this in the setuptools project directory:
+
+       gcc -DGUI=0           -mno-cygwin -O -s -o setuptools/cli.exe launcher.c
+       gcc -DGUI=1 -mwindows -mno-cygwin -O -s -o setuptools/gui.exe launcher.c
+*/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+#include <tchar.h>
+#include <fcntl.h>
+#include <process.h>
+
+
+// http://www.catch22.net/tuts/reducing-executable-size
+// Favour small code
+#pragma optimize("gsy", on)
+
+
+/*if you need it: here is a way to get specific path for different arch... */
+#if __SIZEOF_POINTER__ == 8
+# define ARCH "x64"
+#else
+# define ARCH "i386"
+#endif
+
+/*
+Change these to the *relative* paths your real executable is in
+In the end this is used as
+   <path env><RELATIVE_PATH><wrapper name><EXEC_EXTENSION>
+Slashes are need to be included here!
+<path env> really means one dir below the dir where the wrapper is. Which is
+true for <env>\Scripts and <env>\bin
+*/
+#define RELATIVE_PATH   "\\Library\\inkscape\\"
+#define EXEC_EXTENSION  ".com"
+#define DEBUG 0
+
+
+/* Only change soemthing below where the path is set in the end.. */
+
+int child_pid=0;
+
+
+int fail(char *format, char *data)
+{
+    /* Print error message to stderr and return 2 */
+    fprintf(stderr, format, data);
+    return 2;
+}
+
+
+char *quoted(char *data)
+{
+    int i, ln = strlen(data), nb;
+
+    /* We allocate twice as much space as needed to deal with worse-case
+       of having to escape everything. */
+    char *result = calloc(ln * 2 + 3, sizeof(char));
+    char *presult = result;
+
+    *presult++ = '"';
+    for (nb=0, i=0; i < ln; i++)
+    {
+        if (data[i] == '\\')
+            nb += 1;
+        else if (data[i] == '"')
+        {
+            for (; nb > 0; nb--)
+                *presult++ = '\\';
+            *presult++ = '\\';
+        }
+        else
+            nb = 0;
+        *presult++ = data[i];
+    }
+
+    for (; nb > 0; nb--)        /* Deal w trailing slashes */
+        *presult++ = '\\';
+
+    *presult++ = '"';
+    *presult++ = 0;
+    return result;
+}
+
+
+
+void pass_control_to_child(DWORD control_type)
+{
+    /*
+     * distribute-issue207
+     * passes the control event to child process (Python)
+     */
+    if (!child_pid) {
+        return;
+    }
+    GenerateConsoleCtrlEvent(child_pid, 0);
+}
+
+
+BOOL control_handler(DWORD control_type)
+{
+    /*
+     * distribute-issue207
+     * control event handler callback function
+     */
+    switch (control_type) {
+        case CTRL_C_EVENT:
+            pass_control_to_child(0);
+            break;
+    }
+    return TRUE;
+}
+
+
+int create_and_wait_for_subprocess(char* command) {
+    /*
+     * distribute-issue207
+     * launches child process (Python)
+     */
+    DWORD return_value = 0;
```

Comment:
Not taken, more or less because I'm not sure what kind of error I should chose for the '1' case. So just use "magic numbers"...